internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -781,6 +781,50 @@ func (b *BeatsMonitor) injectMetricsInput(cfg map[string]interface{}, componentI
 				},
 			})
 		}
+
+	}
+
+	shipperHTTPStreams := []interface{}{}
+	// the shipper is listed in componentList, but not componentIDToBinary
+	// iterate over the full component list, adding a monitoring output for every shipper binary.
+	for _, comp := range componentList {
+		if comp.ShipperSpec != nil { // a shipper unit
+			endpoints := []interface{}{prefixedEndpoint(endpointPath(comp.ID, b.operatingSystem))}
+			name := "shipper" // in other beats this is the binary name, but we can hard-code it here.
```

Comment:
You are making the assumption that `shipper` is the only shipper. The specification allows multiple types of shippers to exist. This should be done in a generic way, just checking `comp.ShipperSpec` will not work. You could just use `comp.ShipperSpec.Type`.

Reply:
So, changed it to use `shipperSpec.Spec.Name`. However, did a quick test, and both `Name` and `ShipperType` are set to `"shipper"` for both the running shippers.

Reply:
That works!